torchgen/utils.py

```diff
@@ -209,6 +209,29 @@ def write_sharded(
         num_shards: int,
         base_env: dict[str, Any] | None = None,
         sharded_keys: set[str],
+    ) -> None:
+        self.write_sharded_with_template(
+            filename,
+            filename,
+            items,
+            key_fn=key_fn,
+            env_callable=env_callable,
+            num_shards=num_shards,
+            base_env=base_env,
+            sharded_keys=sharded_keys,
+        )
+
+    def write_sharded_with_template(
+        self,
+        filename: str,
+        template_fn: str,
```

Comment:
nit: `template_fn` -> `template_name`? (it looks like this is the string name of the input template file, not a function)

Reply:
this matches the existing `template_fn` parameter to `write_with_template` and `substitute_with_template`. I agree that it is confusing to overload "fn" to abbreviate both "filename" and "function", and there is a lot of historical precedent for "function", so I will just do a file-wide find-replace.

Reply:
> file-wide find-replace
BC-Lint has informed me that I can't do that, so I will keep it template_fn for consistency :\